### DISCUSSION TITLE
savefluxes mode normalization

### DISCUSSION THREAD

**Discussion by Bovornpratch** [mboquien/cigale#42] [Q&A] [answered]
Hi,

I want to use CIGALE to create mock SEDs of galaxies and AGNs at different redshifts (and parameters) and compare them with survey depths or say put into S/N calculators. 

I took one of the the SEDs I currently have which was fitted with real data and tried using the best-fit parameters to create mock SEDs at different redshifts using the saveflux mode. I was expect CIGALE to output SEDs with gradually fainter fluxes with increasing redshifts.

First as a test, I set the redshift the same as the systematic redshift. So I expecte the results to come out the same as the best-fit model from the pdf-analysis mode. However, I found that the normalization is completely off by ~27 magnitudes but the overall details in the SED are the same (see figure). Therefore, the colors would be the same but not the actual flux values.

The cigale manual suggests that the 
"Note that the extensive properties (e.g., stellar mass and star formation rate) have not been
properly normalized in the results. You might want to normalize by, e.g., stellar mass or
flux, before using these quantities."

Is this issue related? I attach the parameter files with this post.  Is there a way to do correct the normalization or am I missing something.
parameter files :
https://www.dropbox.com/s/nsp8ch65vqiwso4/pcigale.ini?dl=0
https://www.dropbox.com/s/pu3uqi8lt02o7eu/pcigale.ini.spec?dl=0
test seds : https://www.dropbox.com/s/15q1og6m3ycwxw1/2_best_model.fits?dl=0
https://www.dropbox.com/s/ya0oh5cfg0cdbgn/XMM00431_best_model.fits?dl=0

![image](https://user-images.githubusercontent.com/20829556/219308157-df711701-a4c6-4716-91ef-8178d6dc0250.png)

**Comment by mboquien** [comment, marked as answer]
This is indeed because of the mass normalization. By default with `savefluxes` the integral of the mass will be normalized to 1 Msun. What you want to do is to scale the spectra by the value of `best.sfh.integrated` from the best-fit. An offset of 27 mag suggests a log Mstar/Msun of 10.8.

**Reply by Bovornpratch** [reply]
Hi, thanks for the answer. This partially fixes my problem. 

The issue now is how can we obtain the correct normalization factor? In the output model-block file the sfh.integrated is listed as 1 and the integration of the SFH file is also 1. 

In my example I had the parameters from a best-fit where I know the integrated SFH but what about the cases where we just simulate objects such as in the example of the manual file?

**Reply by mboquien** [reply]
The correct scaling when you already have a best-fit is to multiply by `best.sfh.integrated` obtained from that best fit. If you want to do a pure simulation, what I would suggest is to scale `stellar.m_star` to the stellar mass you want for your galaxy and multiply the spectrum (and other extensive properties) by the same factor.
